docs/components/DataList/Web.stories.tsx

```diff
@@ -95,6 +95,14 @@ const Template: ComponentStory<typeof DataList> = args => {
         created: "Created",
       }}
       onLoadMore={nextPage}
+      sorting={{
+        initialState: {
+          key: "label",
+          direction: "desc",
+        },
+        onSort: sorting => console.log(sorting),
+        sortable: ["label", "home"],
+      }}
```

Comment:
This is why I like seeing the component signature first before writing any code.

The fact that you have a prop called `initialState` made me assume that we are tracking a state internally (we are) and that we shouldn't because you risk having the internal and external state being out of sync and causing a lot of headaches for both user and dev.

Also, rule of thumb, the moment you name a prop "initial" in Atlantis, that should be an indicator that you are doing something that's not right.

Reply:
Should I just keep the `sortable` and `onSort` options then? I think it would be useful to have a way to define a initial state for the sorting, I can think of someone wanting to have a "created" column sorted by default, for example.

Reply:
Let's say I have a button somewhere on the page that the DataList doesn't know about. Say it's a feature that saves your filters. Clicking the button does a bunch of like set a filter, search, and also includes updating the state where I determine what sorting the data list should have.

- The user loads the page with the data list
- The DataList renders and it says the initial sorting is nothing
- The user clicks "give me my saved filter"
- How would the data list know you want it to have a different sort?

Reply:
Yeah, I see. In this case, the consumer could set the desired sorting as to the `initialState` and it would update it. Maybe it is not "initial", so we can change the prop name to just "state", or have a way of updating the state from the outside? 🤔

Reply:
yup. states that are configurable by the user and dev should always be outside of the reusable component. We shouldn't track that state, the component should react* to the prop you give it.

_* this explains why react is called react because they want you to write reactive components